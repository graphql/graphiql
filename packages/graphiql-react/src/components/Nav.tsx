/** @jsx jsx */

/**
 *  Copyright (c) 2021 GraphQL Contributors.
 *
 *  This source code is licensed under the MIT license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

import { jsx } from 'theme-ui';
import { PropsWithChildren } from 'react';

import { ReactNodeLike } from '../types';

interface NavProps {
  children: Array<ReactNodeLike>;
}

interface NavItemProps {
  label: string;
  active?: boolean;
}

export const NavItem = ({
  active,
  label,
  children,
}: PropsWithChildren<NavItemProps>) => (
  <button
    aria-label={label}
    sx={{
      color: active ? '#E10098' : '#8c8c8c',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      cursor: 'pointer',
      willChange: 'transform',
      transition: 'transform .2s ease',
      '@media (prefers-reduced-motion: reduce)': {
        transition: 'none',
      },
      ':hover': {
        transform: 'scale(1.1)',
      },
      ':active': {
        transform: 'scale(.95)',
      },
    }}>
    {children}
  </button>
);

export const Nav = ({ children }: NavProps) => {
  return (
    <nav
      sx={{
        display: 'grid',
        gridAutoFlow: 'row',
        gridAutoRows: '2em',
        fontSize: '3em',
      }}>
      {children}
    </nav>
  );
};
