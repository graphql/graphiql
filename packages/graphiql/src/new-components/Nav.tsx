/** @jsx jsx */
import { jsx } from 'theme-ui';
import { PropsWithChildren } from 'react';

const Item = ({ children }: PropsWithChildren<{}>) => (
  <button
    sx={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      cursor: 'pointer',
      transition: '.2s',
      ':hover': {
        transform: 'scale(1.1)',
      },
      ':active': {
        transform: 'scale(.95)',
      },
    }}>
    {children}
  </button>
);

const Nav = () => {
  return (
    <nav
      sx={{
        display: 'grid',
        gridAutoFlow: 'row',
        gridAutoRows: '2em',
        fontSize: '3em',
      }}>
      <Item>{'🐽'}</Item>
      <Item>{'👨‍🌾'}</Item>
      <Item>{'🐝'}</Item>
    </nav>
  );
};

export default Nav;
