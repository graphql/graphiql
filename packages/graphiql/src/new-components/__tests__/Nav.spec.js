import React from 'react';
import renderer from 'react-test-renderer';
import Nav from '../Nav';
import Providers from './fixtures/Providers';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Providers>
        <Nav />
      </Providers>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
