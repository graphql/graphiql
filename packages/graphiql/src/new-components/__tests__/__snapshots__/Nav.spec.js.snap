// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="css-5tspew-Nav"
>
  <button
    className="css-1ijr7ps-Item"
  >
    🐽
  </button>
  <button
    className="css-1ijr7ps-Item"
  >
    👨‍🌾
  </button>
  <button
    className="css-1ijr7ps-Item"
  >
    🐝
  </button>
</nav>
`;
